include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -631,6 +631,12 @@ namespace VectorTools
         if (!cell1->active() && !cell2->active())
           continue;
 
+        // skip foreign cells
+        if (cell1->active() && !cell1->is_locally_owned())
+          continue;
+        if (cell2->active() && !cell2->is_locally_owned())
+          continue;
```

Comment:
How do you check that the two triangulations have this property? Shouldn't there be an assertion that if one is owned by a processor, that the other is as well? Otherwise you will get a result, but it's not going to be what the user wanted.